pkg/encoding/v1/encoding.go

```diff
@@ -481,5 +553,47 @@ func (d *Decoder) Decode(data []byte) (*object.OpenCompose, error) {
 		openCompose.Volumes = append(openCompose.Volumes, ov)
 	}
 
+	// convert secrets
+	for _, secret := range v1.Secrets {
+		var os object.Secret
+		os.Name = string(secret.Name)
+		for _, secData := range secret.Data {
+
+			// If the secret has been provided through a file
+			// and
+			// if it is a relative path
+			// and
+			// the OpenCompose input file is either STDIN or file path
+			// then
+			// convert it to an absolute file path.
+			// This will fail if input file is URL
+			if (secData.File != nil) && !filepath.IsAbs(*secData.File) {
+
+				if in.URL != nil {
+					return nil, fmt.Errorf("Unable to decode secret %v with key %v has a relative file path %v, but is being passed as a URL", secret.Name, secData.Key, secData.File)
```

Comment:
Ack, I will reword the error.

> It will load secrets from a local file, if a path is relative it will be relative to current directory.

I'm not sure if this is a good idea, since most use cases for this will be users putting in URL from their VCS repositories, and the secret file will most likely be present in the VCS.
Wait, but yeah, it would make sense because the secret file will not be committed to the VCS and will be in .gitignore, but might be present locally.

Yep, I think, we should log this message that we will be using the secret from the local path, and use it.

Yep